src/Microsoft.DotNet.Cli.Utils/FrameworkDependencyFile.cs

```diff
@@ -45,6 +46,36 @@ public string GetNetStandardLibraryVersion()
                 ?.Version;
         }
 
+        public bool TryGetMostFitRuntimeIdentifier(IEnumerable<string> runtimeIdentifiers, out string mostFitRuntimeIdentifier)
+        {
+            mostFitRuntimeIdentifier = null;
+
+            IEnumerable<string> supportedRuntimeIdentifiers = runtimeIdentifiers.Distinct().Where(r => IsRuntimeSupported(r));
+
+            if (supportedRuntimeIdentifiers.Any())
+            {
+                mostFitRuntimeIdentifier = supportedRuntimeIdentifiers.Where(r => IsAllOtherNodeOnPathToRoot(r, supportedRuntimeIdentifiers)).First();
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        private bool IsAllOtherNodeOnPathToRoot(string targetRuntimeIdentifierNode, IEnumerable<string> otherRuntimeIdentifierNodes)
```

Comment:
If I'm understanding the object model of `DependencyModel` correctly, then I think it would be clearer to use the following name/signature for this method:

```C#
bool AllAreAncestors(string runtimeIdentifier, IEnumerable<string> potentialAncestors)
```